tasks/files/snapshot.py

```diff
@@ -202,18 +202,83 @@ def lvm_full_report_json():
     return lvm_json
 
 
-def get_snapshot_name(lv_name, prefix, suffix):
-    if prefix:
-        prefix_str = prefix
-    else:
-        prefix_str = ""
+def lvm_get_fs_mount_points(block_path):
+    find_mnt_command = [
+        "findmnt",
+        block_path,
+        "-P",
+    ]
+
+    rc, output = run_command(find_mnt_command)
+    if rc:
+        return None
+
+    output = output.replace('"', "")
+    output = output.replace("\n", "")
+    kv = dict(arg.split("=", 1) for arg in output.split(" ") if arg)
+
+    try:
+        mount_json = json.dumps(kv)
+    except TypeError as error:
+        logger.info(error)
+        return None
+
+    return mount_json
+
+
+def lvm_list_snapset_json(all_lvs, vg_name, lv_name, suffix):
+    lvm_json = lvm_full_report_json()
+    report = lvm_json["report"]
+    vg_dict = dict()
+    fs_dict = dict()
+    lv_list = list()
+    fs_list = list()
+    top_level = dict()
+
+    # Revert snapshots
+    for list_item in report:
+        # The list contains items that are not VGs
+        try:
+            list_item["vg"]
+        except KeyError:
+            continue
+        vg = list_item["vg"][0]["vg_name"]
+        if vg_name and vg != vg_name:
+            continue
+
+        for lv_item in list_item["lv"]:
+            lv_item_name = lv_item["lv_name"]
+            if lv_name and lv_name != lv_item_name:
+                continue
+
+            lv_list.append(lv_item)
+
+        if len(lv_list) > 0:
+            vg_dict[vg] = lv_list
+            lv_list = []
+
+    for lv_list in vg_dict.items():
+        for lv_item in lv_list[1]:
+            block_path = lv_item["lv_path"]
+            fs_mount_points = lvm_get_fs_mount_points(block_path)
+            if fs_mount_points:
+                fs_list.append(fs_mount_points)
+            fs_dict[block_path] = fs_mount_points
+        fs_list = list()
```

Comment:
? the for loop builds `fs_list`, then you reset it here?

Reply:
this is a problem.  will fix.  it isn't working properly for blockdevices that are mounted in multiple mount points.